### PR TITLE
Implement and test debug sequential executor

### PR DESCRIPTION
**Pull request by philippotto**
Adds a new executor type which schedules all jobs immediately, sequentially in the same process.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -121,10 +126,39 @@ def run_square_numbers(executor):
...
+    for exc in get_executors(with_debug_sequential=True):
         run_square_numbers(exc)
 
 
+def get_pid():
+
+    return os.getpid()
+
+
+def test_process_id():
```
Comment (daniel-wer) [inline test.py:138]: Nice test :+1:

File: cluster_tools/__init__.py [hunk 2]
```diff
@@ -184,6 +184,39 @@ def __init__(self, **kwargs):
         WrappedProcessPoolExecutor.__init__(self, **kwargs)
 
 
+class DebugSequentialExecutor(SequentialExecutor):
+    def submit(self, *args, **kwargs):
```
Comment (daniel-wer) [inline cluster_tools/__init__.py:188]: Please add a comment outlining what differentiates this Executor from the SequentialExecutor :)

**Review by daniel-wer — APPROVED** [review]
Nice, I really like the test :+1: 

This is the accompanying issue I think: https://github.com/scalableminds/cluster_tools/issues/99
Did you test that the described debugging use case works with this new executor? The plan probably is to add a config similar to the slurm.yaml or multiprocessing.yaml to make it easy to use this executor during debugging, right?

File: cluster_tools/__init__.py (hunk 2 above)
Comment (philippotto) [reply cluster_tools/__init__.py:203]: good point! i added some docstrings now.

**Comment by philippotto** [comment]
> Did you test that the described debugging use case works with this new executor?

Yes, I tested setting a `breakpoint()` in a job (within `test.py` of the clustertools), which now succeeds with the new executor (the old executors would fail for that) :tada: 

> The plan probably is to add a config similar to the slurm.yaml or multiprocessing.yaml to make it easy to use this executor during debugging, right?

Correct!

File: cluster_tools/__init__.py [hunk 3]
```diff
@@ -179,11 +185,53 @@ def forward_log(self, fut):
...
+    That way, switching between different executors should always work without any problems.
+    """
     def __init__(self, **kwargs):
         kwargs["max_workers"] = 1
         WrappedProcessPoolExecutor.__init__(self, **kwargs)
 
 
+class DebugSequentialExecutor(SequentialExecutor):
+    """
+    Only use for debuggign purposes. This executor does not spawn new processes for its jobs. Therefore,
```
Comment (daniel-wer) [inline cluster_tools/__init__.py:200]:
```suggestion
    Only use for debugging purposes. This executor does not spawn new processes for its jobs. Therefore,
```

**Pull request merged**